imagesgallery/imagesgallery.py

```diff
@@ -55,20 +94,113 @@ def student_view(self, context=None):
         frag.initialize_js('ImagesGalleryXBlock')
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
+    @XBlock.handler
+    def file_upload(self, request, suffix=''):
+        """Handler for file upload to the course assets."""
+        try:
+            from cms.djangoapps.contentstore.views.assets import update_course_run_asset  # pylint: import-outside-toplevel
+        except ImportError:
+            from cms.djangoapps.contentstore.asset_storage_handler import update_course_run_asset  # pylint: import-outside-toplevel
```

Comment:
I imagine there is a reason behind this try catch for this import, can you leave a comment explaining it?

Reply:
Of course! Thanks